contracts/secret-storage/SecretManager.sol

```diff
@@ -356,90 +356,72 @@ contract SecretManager is
     }
 
     function _markStoreAlive(
-        uint256 _secretId,
         uint256 _signTimestamp,
         bytes memory _signature
     ) internal {
-        address enclaveAddress = _verifyStoreAliveSign(_secretId, _signTimestamp, _signature);
-
-        uint256 enclaveIndex = _getSelectedEnclaveIndex(_secretId, enclaveAddress);
-        if(!userStorage[_secretId].selectedEnclaves[enclaveIndex].hasAcknowledgedStore)
-            revert SecretManagerUnacknowledged();
-        if(block.timestamp > userStorage[_secretId].selectedEnclaves[enclaveIndex].lastAliveTimestamp + MARK_ALIVE_TIMEOUT)
-            revert SecretManagerMarkAliveTimeoutOver();
-
-        uint256 endTime;
-        if(block.timestamp > userStorage[_secretId].endTimestamp)
-            endTime = userStorage[_secretId].endTimestamp;
-        else
-            endTime = _signTimestamp;
+        address enclaveAddress = _verifyStoreAliveSign(_signTimestamp, _signature);
 
-        uint256 usdcPayment = (endTime - userStorage[_secretId].selectedEnclaves[enclaveIndex].lastAliveTimestamp) * userStorage[_secretId].sizeLimit * SECRET_STORE_FEE_RATE;
-        userStorage[_secretId].usdcDeposit -= usdcPayment;
-        userStorage[_secretId].selectedEnclaves[enclaveIndex].lastAliveTimestamp = _signTimestamp;
+        ( ,uint256 storageOccupied, , uint256 lastAliveTimestamp, address owner, ) = SECRET_STORE.secretStorage(enclaveAddress);
+        SECRET_STORE.updateLastAliveTimestamp(enclaveAddress, _signTimestamp);
 
-        USDC_TOKEN.safeTransfer(SECRET_STORE.getSecretStoreOwner(enclaveAddress), usdcPayment);
+        uint256 usdcPayment = (block.timestamp - lastAliveTimestamp) * storageOccupied * SECRET_STORE_FEE_RATE;
```

Comment:
can't directly calculate usdc payment based on current occupied storage. Storage occupied can vary over time between last alive and current alive txn. You need to take care of increase and decrease in storage occupied due to new secret and termination of secret.